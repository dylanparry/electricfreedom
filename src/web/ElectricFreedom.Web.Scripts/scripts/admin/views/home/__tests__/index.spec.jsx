import React from 'react';
import createContext from 'react-router-test-context';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import HomeSwitch from '../index';
import Stats from '../stats';

const setup = (pathname = '/') =>
{
  const options = {
    context: createContext({
      location: { pathname, key: null },
    }),
    childContextTypes: {
      router: React.PropTypes.object,
    },
  };
  const wrapper = mount(<HomeSwitch />, options);

  return {
    wrapper,
    snapshot: toJson(wrapper),
  };
};

describe('HomeSwitch', () =>
{
  test('render', () =>
  {
    const { snapshot } = setup();

    expect(snapshot).toMatchSnapshot();
  });

  test('route "/" renders Stats', () =>
  {
    const { wrapper } = setup();

    expect(wrapper.find(Stats).exists()).toBe(true);
  });
});
