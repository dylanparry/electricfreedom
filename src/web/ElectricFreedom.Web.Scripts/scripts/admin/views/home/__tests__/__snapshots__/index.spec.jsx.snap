// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeSwitch render 1`] = `
<homeSwitch>
  <Switch>
    <Route
      component={[Function]}
      computedMatch={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
      location={
        Object {
          "hash": "",
          "key": null,
          "pathname": "/",
          "search": "",
        }
      }
      path="/"
    >
      <stats
        history={
          Object {
            "_listeners": Array [],
            "_notifyListeners": [Function],
            "action": "POP",
            "createHref": [Function],
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": null,
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "key": null,
            "pathname": "/",
            "search": "",
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <p>
          Statistics
        </p>
      </stats>
    </Route>
  </Switch>
</homeSwitch>
`;
